service/notesservice.php

```diff
@@ -262,6 +280,19 @@ private function generateFileName (Folder $folder, $title, $extension, $id) {
         }
     }
 
+    private function getCategoryPath($category) {
+        $category = str_replace('\\', '/', $category);
+        $folders = explode('/', $category);
+        $cleanedFolders = [];
+        foreach($folders as $folder) {
+            $folder = trim($folder);
+            $folder = preg_replace('/^\.+\s*/', '', $folder);
+            if(!empty($folder)) {
+                $cleanedFolders[] = $folder;
+            }
+        }
+        return implode('/', $cleanedFolders);
+    }
```

Comment:
I've just tested removing this method. Then, the core throws a `NotPermittedException`. This currently results in not accepting the whole change (i.e. also content changes) and in creating two (!) empty notes. In order to improve the user-experience (a respective category name could be entered by accident), we should handle this exception.
Of course, this can be easily done, but we have to think about, which behavior is desired:

a) Remove problematic parts from the new category name and perform changing the category to this adjusted name (my current approach).
b) Don't prepare category name; check for Exception by core; and ignore category change if Exception happens.

From the user perspective, I would prefer a). What do you think?